File: hmappy2.cpp
#include<bits/stdc++.h>
long long int hcf(long long int a,long long int b)
{
    long long int rem=a%b;
    if(rem==0)
    return b;
    else
    return hcf(b,rem);
}
main()
{
	int t;
	scanf("%d",&t);
	while(t--)
	{
		long long int n,k,a,b,total=0,lcm,gcd;
		scanf("%lld%lld%lld%lld",&n,&a,&b,&k);
		gcd=hcf(a,b);
        lcm=(a*b)/gcd;
		total=(n/a)+(n/b)-2*(n/lcm);
		if(total>=k)
		printf("Win\n");
		else
		printf("Lose\n");
	}
}

File: hmappy2.java
#include<stdio.h>
int main()
{
	int t;
	scanf("%d",&t);
   while(t--)
	{
		long long int n,a,b,k;
		scanf("%lld%lld%lld%lld",&n,&a,&b,&k);
		long long int w=0,z=1,a1=a,b1=b,t;
	  while (b1 != 0) 
	 {
    t = b1;
    b1 = a1 % b1;
    a1 = t;
    }
     z = (a*b)/a1;
		w=(long long int)(n/a)+(long long int)(n/b)-(long long int)((n/z)*2);
		//printf("%lld ",z);
		if(w>=k)
		printf("Win\n");
		else printf("Lose\n");
	}
	return 0;
}
    
    
    
    